include/unifex/find_if.hpp

```diff
@@ -105,6 +205,70 @@ struct _receiver<Receiver, Func, FuncPolicy>::type {
   }
 };
 
+template <typename Predecessor, typename Receiver, typename Func, typename FuncPolicy>
+struct _operation_state {
+  using receiver_type = receiver_t<Predecessor, Receiver, Func, FuncPolicy>;
+
+  template <typename... Ts>
+  struct find_if_apply {
+    using type = connect_result_t<
+      std::invoke_result_t<
+        typename receiver_type::find_if_helper,
+        typename receiver_type::template unpack_receiver<Receiver>,
+        FuncPolicy,
+        Ts&&...>,
+      typename receiver_type::template unpack_receiver<Receiver>>;
+  };
+  using operation_state_t =
+      typename Predecessor::template value_types<concat_type_lists_unique_t, find_if_apply>::type;
+
+  template<typename InPlaceConstruct>
+  _operation_state(InPlaceConstruct f) : predOp_{f(*this)} {
+  }
+
+  ~_operation_state() noexcept {
+    if (!started_) {
+      innerOp_.destruct();
+    }
```

Comment:
Oh, you're right. It's not that it just needs inverting, this was based on the implementation of let.

The implementation here is a bit different, though, and predOp_ is just initialised inline rather than in a manual lifetime. innerOp is iniitalised and then immediately started and then the receiver functions destroy it anyway, so this would be a double free. So this is simply unnecessary in this implementation.